app/Jobs/GenerateResumeBook.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+// phpcs:disable Generic.CodeAnalysis.UnusedFunctionParameter,SlevomatCodingStandard.Functions.UnusedParameter
+
+namespace App\Jobs;
+
+use App\User;
+use Illuminate\Bus\Queueable;
+use Adldap\Laravel\Facades\Adldap;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+
+class GenerateResumeBook implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * The major (ou) to filter by, or null.
+     *
+     * @var ?string
+     */
+    private $major;
+
+    /**
+     * The cutoff date for resumes.
+     *
+     * @var string
+     */
+    private $resume_date_cutoff;
+
+    /**
+     * The path to the resume book output file. Only valid after handle is complete.
+     *
+     * @var ?string
+     */
+    public $path;
+
+    /**
+     * The datecode in the name of the resume book output file. Only valid after handle is complete.
+     *
+     * @var ?string
+     */
+    public $datecode;
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(?string $major, string $resume_date_cutoff)
+    {
+        $this->major = $major;
+        $this->resume_date_cutoff = $resume_date_cutoff;
+        $this->path = null;
+        $this->datecode = null;
+        $this->tries = 1;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle(): void
+    {
+        $users = User::active()
+            ->whereNotNull('resume_date')
+            ->where('resume_date', '>', $this->resume_date_cutoff)
+            ->get();
+        $filteredUids = $users->pluck('uid');
+
+        if (null !== $this->major) {
+            $majors = $users->mapWithKeys(static function (User $user): array {
+                $search = Adldap::search()->where('uid', '=', $user->uid)->select('uid', 'ou')->first();
```

Comment:
Some users will have multiple entries in Whitepages, some of which won't return their major as the department. This is especially true of student employees. I don't believe there's any order specifically in which they're returned, so you'll need to account for that.

Reply:
Also handle the potential case where someone isn't listed in whitepages. Very few people do it, but people can request to not be listed.